src/services/SyncManager.ts

```
@@ -16,37 +16,51 @@ export class SyncManager<Local> {
     if (since && this.cfg.updatedAtColumn) {
       query = query.gt(this.cfg.updatedAtColumn, since);
     }
-    const { data: rows, error } = await query;
-    if (error) throw error;
-    const remoteRows = rows || [];
+    const { data: remoteRowsData, error: fetchError } = await query;
+    if (fetchError) throw fetchError; // Propagate fetch errors
 
-    // map rows to Local
+    const remoteRows = remoteRowsData || [];
     const remoteItems = (remoteRows as any[]).map(r => this.cfg.mapRow(r));
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)